urllib3/util/url.py

```diff
@@ -192,7 +192,10 @@ def parse_url(url):
             # If given, ports must be integers.
             if not port.isdigit():
                 raise LocationParseError(url)
-            port = int(port)
+            try:
```

Comment:
It's getting pretty much impossible to follow all the comments here, but I'll try to summarize the parse_url change:

port.isdigit() will catch most non-integers, but it will let things such as ² (superscript 2) pass. Therefore the try...except is necessary around int(port), it will catch these. On the other hand, int(port) alone is not enough, because it will allow whitespace, plus and minus prefixes to pass. So the code should be correct as is. The alternative re.match approach I suggested would make the intent and code easier to understand though.